types/underscore/index.d.ts

```diff
@@ -142,7 +142,8 @@ declare module _ {
     type TypeOfDictionary<V> = V extends Dictionary<infer T> ? T : never;
 
     type TypeOfCollection<V> =
-        V extends List<infer T> ? T
+        V extends never ? any
```

Comment:
When I plugged this into my company's solution, I ran into the case where giving an `any` collection to `find` was resulting in `unknown | undefined` instead of `any` and found that `TypeOfCollection<any>` was resulting in `unknown`, which isn't the best for backwards compatibility. I found through experimentation that `extends never` seems to work as a way to determine if a generic type is `any` and added that case to `TypeOfCollection` to help make this more backwards compatible.

`UnderscoreStatic` functions should still error out on a `never` collection because `V` is always constrained with `extends Collection<any>`, and passing a `never` to `_.chain` results in a `_Chain<never, never>` rather than a `_Chain<any, never>`, which is surprising but nice.

Reply:
[This issue](https://github.com/Microsoft/TypeScript/issues/23182) provides an explanation for why `V extends never` evaluates to false for a `V` of `never`.

Reply:
It does feel a bit scary though. I suggest adding some tests against `any` and `never` collections just to rule out that this is going to cause trouble.

Reply:
It's true, I was hoping to come across an issue or PR comment that explained that `any extends never => true` was also definitely expected and correct behavior, but the only reference I found to that was in microsoft/TypeScript#23838 and it admittedly wasn't exactly encouraging.

> A ranges over all types, including null and undefined, but excluding never, any, {} and unknown itself.
>
> For the excluded cases and how they interact with each other, it's not always clear. In some cases it is, but I refer you to things like `any extends never ? true : false;` (try guessing what the result of that one is!).

Reply:
I added the requested tests for `any` and `never` to the OOP and chain test sections in addition to the test that chains an any through filter, reject, and find.

Reply:
I was having a bit of an issue adding `any` tests to each function because TS was inferring stuff from the function iteratee, but I figured that out and added `any` tests for each function as well.